datasets/iwslt2017.py

```diff
@@ -0,0 +1,227 @@
+                            "IWSLT17.TED.tst{}.{}.{}.xml".format(
+                                year, self.config.pair, source
+                            ),
+                        )
+                        for year in years
+                    ],
+                    "target_files": [
+                        os.path.join(
+                            data_dir,
+                            "IWSLT17.TED.tst{}.{}.{}.xml".format(
+                                year, self.config.pair, target
+                            ),
+                        )
+                        for year in years
+                    ],
+                    "split": "test",
+                },
+            ),
+            nlp.SplitGenerator(
+                name=nlp.Split.VALIDATION,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={
+                    "source_files": [
+                        os.path.join(
+                            data_dir,
+                            "IWSLT17.TED.dev2010.{}.{}.xml".format(
+                                self.config.pair, source
+                            ),
+                        )
+                    ],
+                    "target_files": [
+                        os.path.join(
+                            data_dir,
+                            "IWSLT17.TED.dev2010.{}.{}.xml".format(
+                                self.config.pair, target
+                            ),
+                        )
+                    ],
+                    "split": "dev",
+                },
+            ),
+        ]
+
+    def _generate_examples(self, source_files, target_files, split):
+        """ Yields examples. """
+        id_ = 0
+        for source_file, target_file in zip(source_files, target_files):
+            with open(source_file) as sf:
+                with open(target_file) as tf:
+                    for source_row, target_row in zip(sf, tf):
+                        source_row = source_row.strip()
+                        target_row = target_row.strip()
+
+                        if source_row.startswith("<"):
+                            if source_row.startswith("<seg"):
+                                # Remove <seg id="1">.....</seg>
+                                # Very simple code instead of regex or xml parsing
+                                part1 = source_row.split(">")[1]
+                                source_row = part1.split("<")[0]
+                                part1 = target_row.split(">")[1]
+                                target_row = part1.split("<")[0]
+
+                                source_row = source_row.strip()
+                                target_row = target_row.strip()
+                            else:
+                                continue
+
+                        yield id_, {
+                            "source": source_row,
+                            "target": target_row,
+                        }
```

Comment:
If you use the `Translation` feature type defined this way

```python
features=nlp.Features({"translation": nlp.features.Translation(languages=self.config.pair.split("-"))})
```

Then the `yield` statement here would become:

```python
yield id_, {
    "translation": {
        source: source_row,
        target: target_row
    }
}
```

where source and target are defined by
```python
source, target = self.config.pair.split("-")
```

Reply:
Ok done.